Katello/katello
# Fixes #35358 - simplified acs products not removed

#### What are the changes introduced in this pull request?
Products will now be removed from a simplified ACS if the product loses all repositories with populated URLs. This is handled whether the repository's URL is updated to nil or it is removed entirely.

#### Considerations taken when implementing this change?
N/A

#### What are the testing steps for this pull request?

1.  Using hammer, create a simplified ACS and an empty product.
2. Create two repositories and attatch them to the empty product (replacing product id where needed):
```
hammer repository create --name RepoNoURL --product-id 2 --content-type yum
hammer repository create --name RepoURL --url https://jlsherrill.fedorapeople.org/fake-repos/needed-errata/ --product-id 2 --content-type yum
```
3. Attach this product to the ACS (replacing ids where needed). Query the ACS to ensure the product has been added:
```
hammer alternate-content-source update --id 49 --product-ids 2
hammer alternate-content-source info --id 49
```
4. Update the only repo with a URL so that the field is nil. The product should be removed from the ACS.
```
hammer repository update --name RepoURL --product 35358Product --organization-id 1 --url ""
hammer alternate-content-source info --id 49
```
5. Reset the repo and add it back to the ACS:
```
hammer repository update --name RepoURL --product 35358Product --organization-id 1 --url https://jlsherrill.fedorapeople.org/fake-repos/needed-errata/
hammer alternate-content-source update --id 49 --product-ids 2
```
6. Delete the repo so that the product contains no assigned URLs. The product should be removed from the ACS:
```
hammer repository delete --name RepoURL --product 35358Product --organization-id 1
hammer alternate-content-source info --id 49
```
